### PR TITLE
fix: typo in multiple files

### PR DESCRIPTION
**Pull request by thapasusheel**
This PR fix the typos seen in multiple files

```pseudo
continous > continuous
indecies > indicies
posible > possible
thhis > this
reeplace > replace
witth > with
```

### CODE REVIEW

File: runtime/queries/_typescript/highlights.scm [hunk 1]
```diff
@@ -39,11 +39,11 @@
     (identifier) @variable.parameter))
 
 ; (p?: t)
-; (p?: t = 1) // Invalid but still posible to hihglight.
+; (p?: t = 1) // Invalid but still possible to hihglight.
```
Comment (gabydd) [inline runtime/queries/_typescript/highlights.scm:42]: Highlight is also miss spelled in a couple places in this file

File: runtime/tutor [hunk 2]
```diff
@@ -451,7 +451,7 @@
  5. Type u several times to undo your fixes.
  6. Type U (Shift-u) several times to redo your fixes.
 
- --> Fiix the errors on thhis line and reeplace them witth undo.
+ --> Fiix the errors on this line and replace them with undo.
```
Comment (gabydd) [inline runtime/tutor:454]: These errors are meant to be there see the tutor task the snippet is for

**Review by gabydd — CHANGES_REQUESTED** [review]

**Comment by thapasusheel** [comment]
@gabydd  I have made changes as per you have requested. Can you please review it?

**Review by gabydd — APPROVED** [review]

**Comment by uncenter** [comment]
A couple more typos you could fix here (or I can do em in a separate PR):
- https://github.com/helix-editor/helix/blob/master/helix-core/src/transaction.rs#L343 (usally -> usually)
- https://github.com/helix-editor/helix/blob/master/helix-core/src/transaction.rs#L860 (stays ~~inbetween~~ **in between** replacements
- https://github.com/helix-editor/helix/blob/master/helix-term/src/commands.rs#L2243 (continous -> continuous)
- https://github.com/helix-editor/helix/blob/master/helix-term/src/keymap.rs#L267 (upto -> up to)
- https://github.com/helix-editor/helix/blob/master/runtime/queries/nix/injections.scm#L4 (arbitary -> arbitrary)

These aren't comments but "definition" is spelled wrong here? https://github.com/helix-editor/helix/blob/master/runtime/queries/rescript/locals.scm#L6-L7.

**Review by the-mikedavis — CHANGES_REQUESTED** [review]
See https://github.com/helix-editor/helix/pull/8233#pullrequestreview-1618903465

The typos in the highlights are all ok to be fixed but let's avoid touching the rust files here, it will just introduce conflicts for open PRs that touch these blocks. In PRs where I refactor the picker and global search and already change these lines, I have these typos fixed.

**Comment by uncenter** [comment]
Alright, makes sense!

**Comment by thapasusheel** [comment]
> See [#8233 (review)](https://github.com/helix-editor/helix/pull/8233#pullrequestreview-1618903465)
> 
> The typos in the highlights are all ok to be fixed but let's avoid touching the rust files here, it will just introduce conflicts for open PRs that touch these blocks. In PRs where I refactor the picker and global search and already change these lines, I have these typos fixed.

I have made the necessary change can you review it again?

**Comment by the-mikedavis** [comment]
Yeah we run into this piece of advice every now and then even for non-typo-like changes (e.g. renaming structs, making formatting changes, etc.). I'll add a PR with a note about that

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
It'd be nice to fix the typos in the locals queries as well (in this PR or separately): https://github.com/helix-editor/helix/blob/f992c3b5972dbe2432ceb55bc8d47fed912f88bf/runtime/queries/rescript/locals.scm#L6-L7. That typo is causing a small bug in the rescript highlighting.

**Pull request merged**